### ISSUE TITLE
Unknown column name 'STARTINGLOYALTY'

### ISSUE THREAD

**Issue by Baymete** [magefree/mage#4634]
Happens when trying to import a deck. 
java.lang.IllegalArgumentException: Unknown column name 'STARTINGLOYALTY' in table card
Full log attached. I think it might be related to my os language (Turkish) and the "ı" problem. 
As a result, I cannot import any decks.
[mageclient.log](https://github.com/magefree/mage/files/1822784/mageclient.log)

**Comment by LevelX2** [comment]
```
FATAL 2018-03-18 17:39:07,404                                                                                            =>[AWT-EventQueue-0] DeckImporter.importDeck 
java.lang.IllegalArgumentException: Unknown column name 'STARTINGLOYALTY' in table card
	at com.j256.ormlite.table.TableInfo.getFieldTypeByColumnName(TableInfo.java:155)
	at com.j256.ormlite.stmt.RawRowMapperImpl.mapRow(RawRowMapperImpl.java:32)
	at com.j256.ormlite.stmt.StatementExecutor$UserRawRowMapper.mapRow(StatementExecutor.java:636)
	at com.j256.ormlite.stmt.SelectIterator.getCurrent(SelectIterator.java:270)
	at com.j256.ormlite.stmt.SelectIterator.nextThrow(SelectIterator.java:161)
	at com.j256.ormlite.stmt.SelectIterator.next(SelectIterator.java:173)
	at mage.cards.repository.CardRepository.findCardsCaseInsensitive(CardRepository.java:405)
	at mage.cards.repository.CardRepository.findPreferedCoreExpansionCard(CardRepository.java:337)
	at mage.cards.repository.CardRepository.findPreferedCoreExpansionCard(CardRepository.java:330)
	at mage.cards.decks.importer.DecDeckImporter.readLine(DecDeckImporter.java:58)
	at mage.cards.decks.importer.DeckImporter.importDeck(DeckImporter.java:70)
	at mage.client.deckeditor.DeckEditorPanel.btnImportActionPerformed(DeckEditorPanel.java:1017)
	at mage.client.deckeditor.DeckEditorPanel.lambda$initComponents$6(DeckEditorPanel.java:690)
	at javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
	at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
```

**Comment by LevelX2** [comment]
Can you add a deck file that causes that problem? 
Or does really every file cause this error?

**Comment by Baymete** [comment]
Every deck I've tried resulted in the same error. 
Here is one that I tried importing
[ur-unblockable-aggro-2.zip](https://github.com/magefree/mage/files/1822823/ur-unblockable-aggro-2.zip)

**Comment by LevelX2** [comment]
The file is fine, no problem to import it on my system. 

https://garygregory.wordpress.com/2015/11/03/java-lowercase-conversion-turkey/

So the problem could probably be the toLowerCase or toUpperCase on Strings without language parameter. I added those so you can check on next update if it will work.

This could also fix the connect problems on Turkish systems I've seen in the past.

**Comment by JayDi85** [comment]
Try that for fix database corruption/upgrade:
 * Open xmage folder;
 * Search all `*cards.h2*` files and delete it (from client and from server's folders);
 * Restart client and connect to any server like xmage.de ;
 * Try to import again and report results.

**Comment by Baymete** [comment]
@JayDi85 tried as you described but it didn't work. Connecting to server does not work also and I'm pretty sure it is related to Turkish characters.
[mageclient.log](https://github.com/magefree/mage/files/1825652/mageclient.log)

**Comment by JayDi85** [comment]
Found [turkish topic here](https://www.slightlymagic.net/forum/viewtopic.php?f=70&t=16424) -- it's can help only english OS language.
> If you use Windows in Turkish. You get that error. If you add english language (not keyboard or localization language but MENU language) you can connect.
> Or you can install fresh english OS to your system.

You can try to setup java default local settings. Use [instructions from here]( https://stackoverflow.com/a/31529213/1276632):
 * Open launcher settings and add java command: -Duser.language=en-US

**Comment by JayDi85** [comment]
Another one things -- change locale for custom app by that tool: [Locale Emulator](http://pooi.moe/Locale-Emulator/) (I don't use that but someone using it for other games and apps).

**Comment by zeflap** [comment]
@JayDi85 I've seen this bug in many Java-based game projects, http://mattryall.net/blog/2009/02/the-infamous-turkish-locale-bug has been helpful to many.

**Comment by Zerrisx** [comment]
Do we have any plans to make XMage changes to handle this?

**Comment by JayDi85** [comment]
It's already handle it in most cases. I fixes last missing code by e69a021c7159ed9711b8988c2ccbf87671a21b14

For developers -- IntelliJ IDEA can find all problems with `toLowerCase` and `toUpperCase` without `Local.ENGLISH` param. Inspection settings example:

![shot_200703_015436](https://user-images.githubusercontent.com/8344157/86412614-6b70de80-bcd0-11ea-844e-498a12e30273.png)
